tests/TranslatableTest.php

```diff
@@ -98,7 +98,7 @@ public function it_can_save_a_translated_attribute()
     public function it_can_set_translated_values_when_creating_a_model()
     {
         $model = TestModel::create([
-            'name' => ['en' => 'testValue_en'],
+            'name' => 'testValue_en',
```

Comment:
Why did this change? Will the tests still pass with the old value?

Reply:
So, the problem is that at the moment you can save a model with all possible translations directly by passing a translation map (e.g., `['en' => 'testValue_en']`). However, in case you are dealing with a JSON field we cannot differentiate whether the passed value is a translation map or the object itself and the second part of this check will always be true:
https://github.com/spatie/laravel-translatable/blob/02ba5da0587df044ca57f37f19ca6112a02e3a8a/src/HasTranslations.php#L30 

That's why I came up with that locale check in the first place to differentiate between those both. 

Though, having a second thought we could also extend the `$translatable` attribute and use that as a differentiator:
```php
public $translatable = ['string_field', 'json_field' => [ 'type' => 'json' ]];
```

What do you think?

Reply:
Having a third thought... as long as we allow arrays that contain translations for multiple locales to be passed to an attribute we cannot properly support JSON fields. There is no chance at all to differentiate between the both cases. So I think JSON support would definitely be a breaking change 😞